driver/csiplugin/connectors/rest_v2.go

```diff
@@ -709,22 +765,64 @@ func (s *SpectrumRestV2) DeleteBucketKeys(ctx context.Context, bucket string) er
 	return nil
 }
 
-func (s *SpectrumRestV2) CreateS3CacheFileset(ctx context.Context, filesystemName string, filesetName string, mode string, opts map[string]interface{}, bucketInfo map[string]string, scheme string) error {
+func (s *SpectrumRestV2) DeleteNodeMappingAFMWithCos(ctx context.Context, exportMapName string) error {
 	loggerID := utils.GetLoggerId(ctx)
-	klog.V(4).Infof("[%s] rest_v2 CreateS3CacheFileset. filesystem: %s, fileset: %s, mode: %s, opts: %v, scheme: %s", loggerID, filesystemName, filesetName, mode, opts, scheme)
+	klog.V(4).Infof("[%s] rest_v2 DeleteNodeMappingAFMWithCos", loggerID)
+
+	deleteExportMapNameURL := "scalemgmt/v2/nodes/afm/mapping/" + exportMapName
+	deleteExportMapNameResponse := DelExportMappingResponse{}
+	err := s.doHTTP(ctx, deleteExportMapNameURL, "DELETE", &deleteExportMapNameResponse, nil)
+	if err != nil {
+		klog.Errorf("[%s] Failed to delete exportMap for the exportMapName %s, error: %v", loggerID, exportMapName, err)
+		if strings.Contains(deleteExportMapNameResponse.Status.Message, "Invalid value in 'mappingName'") { // error as no exportMap exists
+			klog.V(6).Infof("[%s] exportMap would have been deleted. So returning success %v", utils.GetLoggerId(ctx), err)
+			return nil
+		}
+		return err
+	}
+
+	err = s.isDelJobRequestAccepted(ctx, deleteExportMapNameResponse, deleteExportMapNameURL)
+	if err != nil {
+		klog.Errorf("[%s] The delete exportMap request is not accepted for processing for the exportMapName %s, error: %v", loggerID, exportMapName, err)
+		return err
+	}
+
+	err = s.WaitForDelJobCompletion(ctx, deleteExportMapNameResponse.Status.Code, deleteExportMapNameResponse.Jobs[0].JobID)
+	if err != nil {
+		klog.Errorf("[%s] Failed to delete keys for the exportMap %s, error: %v", loggerID, exportMapName, err)
+		return err
+	}
+	return nil
+}
+
+func (s *SpectrumRestV2) CreateS3CacheFileset(ctx context.Context, filesystemName string, filesetName string, mode string, opts map[string]interface{}, bucketInfo map[string]string, exportMapName string, parsedEndpointURL *url.URL) error {
```

Comment:
Is this tested with at least 2 cache volumes using 2 different set of keys (e.g. one with readOnly and other with readWrite access) for a single bucket? the read/write access to bucket should be as per the keys

Reply:
As mentioned by Badri, this will be covered by @saurabhwani5 during testing